### ISSUE TITLE
Transactions failing to complete

### ISSUE THREAD

**Issue by calfer** [ethereum/mist#1369]
#### System information
``````
Version: `0.8.7`
OS & Version: MacOS Sierra
Node type: `eth/geth(default)`
``````

#### Please check the **already existing** issues to keep duplicates at a minimum.
Furthermore several **work-arounds** have been collected in the [Mist-Troubleshooting-Guide](https://github.com/ethereum/wiki/wiki/Mist-Troubleshooting-Guide).

If possible add the following to your report:

- Screenshots

- Check the console, of Mist (`CTRL/CMD + ALT + i`) and take a screenshot

- Log files 
  - go to `menu -> accounts -> backup -> application data`
  - zip and upload `node.log` and all other `node.log.X` files 
 
Twice today when sending ETH I received a message in a yellow or orange box in the lower right corner of the window that said the transaction could not be completed.  Both of these transactions were initiated immediately following a previous transaction.  The previous transactions in both instances completed successfully whereas the transaction immediately following each successful transaction were reported as having failed.  Interestingly, the first failed transaction actually ended up going through, debiting my account and crediting the account at the recipient end.  The second failed transaction was simply a transfer between account within the wallet application.  Both transactions hit the blockchain.  The one that actually did fail still shows as pending on etherscan.  Neither of the two transactions that were reported as failures show up in the ledger in the wallet application even though one of them completed successfully despite having been reported as failing.

A subsequent attempt to send the transaction that actually did fail resulted in another failed transaction.

[Archive.zip](https://github.com/ethereum/mist/files/562904/Archive.zip)
![screen shot 2016-10-31 at 4 20 20 pm](https://cloud.githubusercontent.com/assets/19778790/19872183/650bb79c-9f86-11e6-83dd-07db5978d878.png)

**Comment by calfer** [comment]
As it turns out, later on in the day yesterday the failed transaction actually was confirmed and written to the blockchain.  As mentioned in my original post I re-sent the failed transaction.  In the end, my account was debited twice.  Next time I will not take the message content at face value.  The wallet ledger still does not show the failed transactions.

**Comment by frozeman** [comment]
The problem is sometimes transactions persist in the tx pool of the node. E.g. if by accident you wrote a tx with a higher nonce, this tx stays in the pool and once you send a mother tx which fill the gap between this and the higher nonced tx, it will also execute the older one.

@obscuren we should probably do something about that and discard tx, if the nonce is +2 as the current nonce?

**Comment by calfer** [comment]
I just attempted another transaction today but it too failed.  What is the recommended corrective action to take as related specifically to my configuration?  I sincerely appreciate the attention and responsiveness of the Ethereum user and development community.

**Comment by evertonfraga** [comment]
Duplicate of https://github.com/ethereum/mist/issues/1408.
